cmd/ecs-deploy/task_def.go

```diff
@@ -430,28 +503,20 @@ func taskDefFunction(cmd *cobra.Command, args []string) error {
 	serviceNameShort := serviceNameParts[0]
 
 	// Confirm the image exists
-	ecrImage, errECRImage := NewECRImage(imageURI)
+	ecrImage, errECRImage := NewECRImage(imageURI, serviceECR)
 	if errECRImage != nil {
 		quit(logger, nil, fmt.Errorf("unable to recognize image URI %q: %w", imageURI, errECRImage))
 	}
 	imageIdentifier := ecr.ImageIdentifier{}
-	imageIdentifier.SetImageTag(ecrImage.ImageTag)
+	if ecrImage.IsDigestBased {
+		imageIdentifier.SetImageDigest(ecrImage.Digest)
+	}
+	if ecrImage.IsTagBased {
+		imageIdentifier.SetImageTag(ecrImage.Tag)
+	}
```

Comment:
Down here I'd expect:

```suggestion
	if ecrImage.ImageURIByDigest != nil {
		imageIdentifier.SetImageDigest(ecrImage.Digest)
	} else if ecrImage.ImageURIByTag != nil {
		imageIdentifier.SetImageTag(ecrImage.Tag)
	} else {
	    quit(logger, nil, fmt.Error("Image Tag or Digest required, neither was parsed."))
	}
```

Reply:
done!